src/DataStructures.jl

```diff
@@ -14,11 +14,11 @@ module DataStructures
                  ReverseOrdering, Reverse, Lt,
                  isless,
                  union, intersect, symdiff, setdiff, issubset,
-                 find, searchsortedfirst, searchsortedlast, endof, in
+                 find, searchsortedfirst, searchsortedlast, endof, in, produce
 
 
     export Deque, Stack, Queue
-    export deque, enqueue!, dequeue!, update!,iter
+    export deque, enqueue!, dequeue!, update!, forward_iter, backward_iter
```

Comment:
see the discussion in the corresponding issue. I was using `backward_iter` as the name from the discussion there, happy to change to `reverse_iter`, though for reasons described below I think this may not be the best idea.

Currently, the default iterator uses LIFO order for stacks and FIFO for queues, whereas `forward_iter` is always FIFO and `backward_iter` is always LIFO.  I think it makes sense for the default iterator to use the ordering that makes sense for the particular data structure whereas the separate functions for `forward_iter` and `backward_iter` should be consistent across data structures (and the term backward versus reverse in the case would make more sense). Let me know what you think about that